rust/datafusion/src/sql/planner.rs

```diff
@@ -114,7 +115,17 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         alias: Option<String>,
     ) -> Result<LogicalPlan> {
         let set_expr = &query.body;
-        let plan = self.set_expr_to_plan(set_expr, alias)?;
+        let ctes: HashMap<String, Query> = query
+            .clone()
+            .with
+            .map(|with| {
+                with.cte_tables
```

Comment:
This needs to be converted to a logical plan here already and only allow backward references